unit-tests/shared/src/test/scala/javalib/net/SocketTest.scala

```diff
@@ -138,19 +139,74 @@ class SocketTest {
     }
   }
 
+  /* The Oracle documentation for Socket method traffic class in both
+   * Java 8 and 17 describe setTrafficClass as providing a hint
+   * and that a setTrafficClass followed by a getTrafficClass of the
+   * might not return the same value.
+   *
+   * But wait! It gets better.
+   *
+   * The setTrafficClass and getTrafficClass methods both use
+   * the StandardSystemsOption IP_TOS field. Both Java 8 and 17
+   * describe this field:
+   *   "The behavior of this socket option on a stream-oriented socket,
+   *   or an IPv6 socket, is not defined in this release."
+   *
+   * This file is testing Sockets() which means stream (TCP) sockets. Strike 1.
+   * The default underlying protocol is now IPv6. Strike 2.
+   *
+   * In the general case and inherent design, this test is bogus.
+   * It is executed only in cases where it has historically passed.
+   * Some day they may break and need to be skipped.
+   *
+   * Other cases are silently skipped, so as to not cause anxiety
+   * over a 'normal' situation.
+   */
   @Test def trafficClass(): Unit = {
-    // When execution on Windows with Java 17 trafficClass is not set.
-    // s.getTrafficClass returns 0 instead of 0x28
-    assumeFalse(
-      "Skipped due to unexpected behaviour in JDK 17 on Windows",
-      Platform.isWindows && Platform.executingInJVMOnJDK17
-    )
-    val s = new Socket()
-    try {
-      s.setTrafficClass(0x28)
-      assertEquals(s.getTrafficClass, 0x28)
-    } finally {
-      s.close()
+
+    val prop = System.getProperty("java.net.preferIPv4Stack")
+    val useIPv4 = (prop != null) && (prop.toLowerCase() == "true")
```

Comment:
Not really important but we can use `sys.SystemProperties.preferIPv4Stack.value` here. This hint can be ignored here.

Reply:
I can retrace the sys.SystemProperties.preferIPv4Stack path again.  I usually like single points
of implementation.  Either I found the sys.systemProperties after I wrote this.   My memory
is that there is a subtlety about the time the (lazy?) val is evaluated. The present code
is pretty sure to evaluate when the first Socket is accessed.

Reply:
I like to do thing right the first time, but this whole initialization is subject to "appropriate precision". 
That is, it is almost certain that likely change will overwhelm & obliterate putting too fine a point on
this initialization in particular.

Thank you for the merge.